### PR TITLE
CAS-399 Ensure status field in NioCtx is always defined.

### PR DESCRIPTION
**Pull request by chriswldenyer**
Ensure the status field in the NioCtx gets defined when the Bio is created. 
This change also reduces some code duplication by adding a constructor for the Bio.

### CODE REVIEW

**Review by paulyoong — APPROVED** [review]

File: mayastor/src/bdev/nexus/nexus_io.rs [hunk 1]
```diff
@@ -87,6 +87,13 @@ impl Bio {
         unsafe { Bdev::from((*self.0).bdev) }
     }
 
+    pub fn new(pio: *mut spdk_bdev_io, in_flight: i8) -> Bio {
```
Comment (jonathan-teh) [inline mayastor/src/bdev/nexus/nexus_io.rs:90]: Should have a comment for each function.

Comment (chriswldenyer) [reply mayastor/src/bdev/nexus/nexus_io.rs:90]: Ok, I'll fix that.

**Review by jonathan-teh — APPROVED** [review]
Looks good although I wonder if it should be fixed in the caller, io_submit, instead.

**Comment by chriswldenyer** [comment]
> Looks good although I wonder if it should be fixed in the caller, io_submit, instead.

Maybe, but the problem with that is that the in_flight value varies per io_type, so the caller would have to take the io_type into account. This way the individual io functions look after that.

**Comment by jonathan-teh** [comment]
> Maybe, but the problem with that is that the in_flight value varies per io_type, so the caller would have to take the io_type into account. This way the individual io functions look after that.

True, but I should have been more precise, I meant the initialisation of the status field could be fixed in the caller.

**Pull request merged**